### PR TITLE
Change the implementation for Syntax Changes v2.

### PR DESCRIPTION
**Pull request by hasithajayasundara**
## Automation tests
 - Unit tests 
   > Code coverage information
 - Integration tests
   > Details about the test cases and coverage

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes/no
 - Ran FindSecurityBugs plugin and verified report? yes/no
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes/no

### CODE REVIEW

File: sonarqube67/sonarqube_connector.bal [hunk 1]
```diff
@@ -224,155 +145,100 @@ The code MUST be immediately reviewed. "}
...
-    } catch (error dataError) {
-        return dataError;
-    }
-    var convertedValue = <int>value;
-    match convertedValue {
-        int val => return val;
-        error castError => return castError;
+        string value = getMetricValue(projectKey, sonarqubeConnector, ISSUE_MINOR);
+        convertedValue =? <int>value;
+    } catch (error err) {
```
Comment (muthulee) [inline sonarqube67/sonarqube_connector.bal]: Even if you don't do try and catch  the error will be returned.

**Review by dimslee — CHANGES_REQUESTED** [review]

File: sonarqube67/sonarqube_transformers.bal [hunk 2]
```diff
@@ -16,7 +16,7 @@
 // under the License.
 //
 
-package sonarqube;
+package sonarqube6;
 
 transformer <json projectDetails, Project project> getProjectDetails() {
```
Comment (Biruntha) [inline sonarqube67/sonarqube_transformers.bal]: Can you convert transformer to function ?

**Review by Biruntha — APPROVED** [review]

File: tests/test.bal [hunk 3]
```diff
@@ -16,171 +16,335 @@
...
-
-    //Get project
-    sonarqube:Project project = {};
-    var projectDetails = sonarqubeEP -> getProject(args[2]);
+@test:Config {
+    groups:["network-calls"]
+}
+function testGetProject () {
+    log:printInfo("sonarqubeEndpoint -> getProject()");
+    var projectDetails = sonarqube -> getProject("siddhi");
```
Comment (Biruntha) [inline tests/test.bal]: Can we get the project key from constant file as you did in other test case?

**Comment by Biruntha** [comment]
1. Can you add .ballerina folder/file inside the project? otherwise we will get the error "ballerina: cannot find package 'tests'"

**Review by dimslee — APPROVED** [review]

File: sonarqube6/sonarqube_connector.bal [hunk 4]
```diff
@@ -0,0 +1,339 @@
...
+                                    project = sonarqubeProject;
+                                    break;
+                                }
+                            } else {
+                                return endpointErrors;
+                            }
+                        }
+                        http:HttpConnectorError err => return err;
+                    }
+                    count = count + 1;
```
Comment (vlgunarathne) [inline sonarqube6/sonarqube_connector.bal]: We can use "count +=1" also.

File: sonarqube6/sonarqube_connector.bal [hunk 5]
```diff
@@ -0,0 +1,339 @@
...
+        }
+        http:HttpConnectorError err => return err;
+    }
+}
+
+@Description {value:"Get complexity of a project."}
+@Return {value:"complexity:Returns complexity of a project.Complexity calculated based on the number of paths through
+ the code."}
+@Return {value:"err: Returns error if an exception raised in getting project complexity."}
+public function SonarQubeConnector::getComplexity(string projectKey) returns (int|error) {
```
Comment (vlgunarathne) [inline sonarqube6/sonarqube_connector.bal]: Needs to have @Param for "projectKey"

File: sonarqube6/sonarqube_connector.bal (hunk 4 above)
Comment (hasithajayasundara) [reply sonarqube6/sonarqube_connector.bal]: Changed this as suggested.

File: sonarqube6/sonarqube_connector.bal (hunk 5 above)
Comment (hasithajayasundara) [reply sonarqube6/sonarqube_connector.bal]: Added.

**Pull request merged**